app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java

```diff
@@ -244,18 +239,32 @@ protected void onCreate(Bundle savedInstanceState) {
         mAssistantHandler = new Handler(mAssistantThread.getLooper());
 
         try {
-            Log.d(TAG, "creating voice hat driver");
-            mVoiceHat = new VoiceHatDriver(I2S_BUS, DAC_TRIGGER_GPIO, AUDIO_FORMAT_STEREO);
-            mVoiceHat.registerAudioInputDriver();
-            mVoiceHat.registerAudioOutputDriver();
-            mButton = new Button(BUTTON_PIN, Button.LogicState.PRESSED_WHEN_LOW);
+            if (AUDIO_USE_I2S_VOICEHAT_IF_AVAILABLE) {
+                PeripheralManagerService pioService = new PeripheralManagerService();
+                List<String> i2sDevices = pioService.getI2sDeviceList();
+                if (i2sDevices.size() > 0) {
+                    try {
+                        Log.d(TAG, "creating voice hat driver");
+                        mVoiceHat = new VoiceHatDriver(
+                                BoardDefaults.getI2SDeviceForVoiceHat(),
+                                BoardDefaults.getGPIOForVoiceHatTrigger(),
+                                AUDIO_FORMAT_STEREO
+                        );
+                        mVoiceHat.registerAudioInputDriver();
+                        mVoiceHat.registerAudioOutputDriver();
+                    } catch (IllegalStateException e) {
+                        Log.e(TAG, "Unsupported board, falling back on default audio device:", e);
+                    }
+                }
+            }
+            mButton = new Button(BoardDefaults.getGPIOForButton(), Button.LogicState.PRESSED_WHEN_LOW);
             mButton.setDebounceDelay(BUTTON_DEBOUNCE_DELAY_MS);
             mButton.setOnButtonEventListener(this);
             PeripheralManagerService pioService = new PeripheralManagerService();
-            mLed = pioService.openGpio(LED_PIN);
+            mLed = pioService.openGpio(BoardDefaults.getGPIOForLED());
             mLed.setDirection(Gpio.DIRECTION_OUT_INITIALLY_LOW);
         } catch (IOException e) {
-            Log.d(TAG, "error creating voice hat driver:", e);
+            Log.d(TAG, "error configuring peripherals:", e);
```

Comment:
Standardize on a log level for your exceptions. ERROR is used below, I would recommend WARNING across the board.